src/wallet/mod.rs

```diff
@@ -140,6 +166,39 @@ impl WalletManager {
         Ok(())
     }
 
+    /// Encrypt the wallet file with a password.
+    pub async fn encrypt(&self, password: &str) -> anyhow::Result<()> {
+        let file = read(&self.path).await?;
+        let encryptor = age::Encryptor::with_user_passphrase(Secret::new(password.to_owned()));
+        let mut encrypted = vec![];
+        let mut writer = encryptor.wrap_output(&mut encrypted).unwrap();
+        writer.write_all(&file)?;
+        writer.finish()?;
+        write(&self.path, &encrypted).await?;
+        Ok(())
+    }
+
+    /// Decrypt the wallet file.
+    pub async fn decrypt(&self, password: &str, only_read: bool) -> anyhow::Result<Vec<u8>> {
+        let file = read(&self.path).await?;
+        let decrypted = {
+            let decryptor = match age::Decryptor::new(file.as_slice())? {
+                age::Decryptor::Passphrase(d) => d,
+                _ => unreachable!(),
```

Comment:
Maybe write a note inside `unreachable!()` on why it's unreachable? As someone who doesn't know how `age` works I don't really know what is happening here.